core/Cargo.toml

```diff
@@ -13,9 +13,15 @@ edition = "2018"
 cfg-if = "1.0.0"
 smallvec = "1.6.1"
 petgraph = { version = "0.6.0", optional = true }
-thread-id = { version = "4.0.0", optional = true }
 backtrace = { version = "0.3.60", optional = true }
 
+[target.'cfg(not(target_family = "wasm"))'.dependencies]
```

Comment:
This condition is different so it doesn't use thread-id for any wasm targets (cause it won't compile). But for wasm-32-unknown is uses js-sys

Reply:
I don't think we should be specifically doing this: if the deadlock detector isn't supported on a particular platform then we should just fail to compile. In the future it might be useful to make the `thread_id` crate work on wasm-wasi and emscripten targets.

Reply:
For wasi target it does work, it just won't have any backtrace or thread/worker name fields. Maybe backtrace works on wasi (was completely empty for wasm) so maybe I could re-add that field

Reply:
I would prefer if the changes in this PR were only for wasm32-unknown-unknown. If additional changes are needed for wasi, these should probably go as a PR in the thread_id crate.

Reply:
done 👍